test/extended/apibinding/apibinding_utils.go

```diff
@@ -0,0 +1,85 @@
+package apibinding
+
+import (
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/kcp-dev/kcp-tests/test/extended/util"
+)
+
+// APIBinding struct defination
+type APIBinding struct {
+	APIVersion string `json:"apiVersion"`
+	Kind       string `json:"kind"`
+	Metadata   struct {
+		Name string `json:"name"`
+	} `json:"metadata"`
+	Spec struct {
+		Reference struct {
+			Workspace struct {
+				Path       string `json:"path"`
+				ExportName string `json:"exportName"`
+			} `json:"workspace"`
+		} `json:"reference"`
+	} `json:"spec"`
+}
+
+// ABOption use function option mode to change the default values of Apibinding attributes
+type ABOption func(*APIBinding)
+
+// SetAPIBindingName replace the default value of APIBinding name
+func SetAPIBindingName(name string) ABOption {
+	return func(a *APIBinding) {
+		a.Metadata.Name = name
+	}
+}
+
+// SetAPIBindingReferencePath replace the default value of APIBinding workspace reference path
+func SetAPIBindingReferencePath(path string) ABOption {
+	return func(a *APIBinding) {
+		a.Spec.Reference.Workspace.Path = path
+	}
+}
+
+// SetAPIBindingReferenceExportName replace the default value of APIBinding workspace reference exportName
+func SetAPIBindingReferenceExportName(exportName string) ABOption {
+	return func(a *APIBinding) {
+		a.Spec.Reference.Workspace.ExportName = exportName
+	}
+}
+
+// NewAPIBinding create a new customized APIBinding
+func NewAPIBinding(opts ...ABOption) APIBinding {
+	defaultAPIBinding := APIBinding{
+		APIVersion: "apis.kcp.dev/v1alpha1",
+		Kind:       "APIBinding",
+		Metadata: struct {
+			Name string `json:"name"`
+		}{Name: "e2e-apibinding-" + exutil.GetRandomString()},
+		Spec: struct {
+			Reference struct {
+				Workspace struct {
+					Path       string `json:"path"`
+					ExportName string `json:"exportName"`
+				} `json:"workspace"`
+			} `json:"reference"`
+		}{},
+	}
+	for _, o := range opts {
+		o(&defaultAPIBinding)
+	}
+	return defaultAPIBinding
+}
+
+// Create APIBinding CR
+func (apb *APIBinding) Create(k *exutil.CLI) {
```

Comment:
Similarly too, APIBinding has existing lib, for example https://github.com/kcp-dev/kcp/blob/main/test/e2e/apibinding/apibinding_test.go uses lib to do `Create` operation:
```
		_, err = kcpClusterClient.ApisV1alpha1().APIBindings().Create(logicalcluster.WithCluster(ctx, consumerWorkspace), apiBinding, metav1.CreateOptions{})
```
Is it worth for us to reference too to see if we can reuse the existing lib?

Reply:
Yeah the origin existing libs maybe a bit complex for us. I'll try to reuse it and make it easily to use for us. Thanks for the details!